python/pyspark/sql/types.py

```diff
@@ -1447,6 +1447,26 @@ def _from_numpy_type(nt: "np.dtype") -> Optional[DataType]:
     return None
 
 
+def _from_numpy_type_to_java_type(nt: "np.dtype", gateway: JavaGateway) -> Optional[JavaClass]:
```

Comment:
You can actually add this as a `NumpyArrayConverter`'s class attribute

Reply:
Did you mean an instance method?